libs/output-mapping/tests/Storage/TableStructureModifierTest.php

```diff
@@ -0,0 +1,152 @@
+    {
+        $source = $this->createMock(MappingFromProcessedConfiguration::class);
+        $source->method('getPrimaryKey')->willReturn(array_merge(
+            $this->destinationTableInfo->getPrimaryKey(),
+            ['Id', 'Name'],
+        ));
+        $source->method('getColumns')->willReturn($this->destinationTableInfo->getColumns());
+        $source->method('getMetadata')->willReturn([]);
+        $source->method('getColumnMetadata')->willReturn([]);
+
+        $this->modifier->updateTableStructure(
+            $this->destinationBucket,
+            $this->destinationTableInfo,
+            $source,
+            $this->destination,
+        );
+
+        $newTable = $this->clientWrapper->getTableAndFileStorageClient()->getTable($this->firstTableId);
+
+        $this->assertCount(count($this->destinationTableInfo->getPrimaryKey()) + 2, $newTable['primaryKey']);
+        $this->assertEquals(['Id', 'Name'], $newTable['primaryKey']);
+    }
+
+    #[NeedsTestTables]
+    public function testErrorUpdateTableStructureBadPK(): void
+    {
+        $source = $this->createMock(MappingFromProcessedConfiguration::class);
+        $source->method('getPrimaryKey')->willReturn(array_merge(
+            $this->destinationTableInfo->getPrimaryKey(),
+            ['invalidPK'],
+        ));
+        $source->method('getColumns')->willReturn($this->destinationTableInfo->getColumns());
+        $source->method('getMetadata')->willReturn([]);
+        $source->method('getColumnMetadata')->willReturn([]);
+
+        $this->modifier->updateTableStructure(
+            $this->destinationBucket,
+            $this->destinationTableInfo,
+            $source,
+            $this->destination,
+        );
+
+        $expectedErrorMessage = 'Error changing primary key of table in.c-testUpdateTableStructureBadPKTest.test1:';
+        $expectedErrorMessage .= ' Primary key columns "invalidPK" not found in "Id, Name, foo, bar"';
+        $this->assertTrue($this->testHandler->hasWarning($expectedErrorMessage));
+    }
+
+    #[NeedsTestTables]
+    public function testErrorUpdateTableStructureEmptyColumnName(): void
+    {
+        $source = $this->createMock(MappingFromProcessedConfiguration::class);
+        $source->method('getPrimaryKey')->willReturn($this->destinationTableInfo->getPrimaryKey());
+        $source->method('getColumns')->willReturn(array_merge(
+            $this->destinationTableInfo->getColumns(),
+            [''],
+        ));
+        $source->method('getMetadata')->willReturn([]);
+        $source->method('getColumnMetadata')->willReturn([]);
+
+        $this->expectException(ClientException::class);
+        $this->expectExceptionMessage('Invalid parameters - name: This value should not be blank.');
+        $this->modifier->updateTableStructure(
+            $this->destinationBucket,
+            $this->destinationTableInfo,
+            $source,
+            $this->destination,
+        );
```

Comment:
ještě by bylo dobrý otestovat ten rollback změny primárního klíče, na to je te teda asi jedinej rozumnej case, - že se má udělat PK na sloupci ve kterým jsou duplicity (což je teda předopklůádá do tabulky nahrát CSV)